src/code/z_common_data.c

```diff
@@ -18,6 +18,6 @@ void SaveContext_Init(void) {
     gSaveContext.nextDayTime = 0xFFFF;
     gSaveContext.skyboxTime = 0;
     gSaveContext.dogIsLost = true;
-    gSaveContext.nextTransition = 0xFF;
+    gSaveContext.nextTransitionType = 0xFF;
```

Comment:
Would it make sense to have an enum value for 0xFF? Maybe _NONE since this seems to be the default for not having a specific transition type?

Reply:
oh yeah, `nextTransitionType` is kinda better described as an override to take over from what is in the entrance table. I was thinking about naming the variable as such, but wasnt sure if I wanted to commit to it.

the distinction is important I think, cause 0xFF only means something to the "override" variable, it doesnt correspond to an actual transition type that the transition system can do anything about. 

so yea `transitionTypeOverride` and `TRANS_OVERRIDE_NONE` are a possibility, but im not 100% sure.
It can be left as `nextTransistionType` and have a define of `TRANS_TYPE_NONE`, but its slightly misleading cause its not the exact same kind of value as the other `TRANS_TYPE_`s

Reply:
its also *only* for the case where a new scene is loading, so for the `TRANS_TRIGGER_END` case. maybe the name should reflect that as well. cause you end up with code like this where it sets the outgoing transition to one type, and the override for the other side of the transition to another
```
        globalCtx->transitionTrigger = TRANS_TRIGGER_START;
        globalCtx->transitionType = TRANS_TYPE_FADE_WHITE;
        gSaveContext.nextTransitionType = TRANS_TYPE_FADE_WHITE_SLOW;
```
but then again maybe it being called "next" is already getting that point across well, not sure

Reply:
Maybe `nextTransitionEndType`/`transitionEndType` ? I'm not a fan though :/

The purpose is to set the transition type for the second half of a scene transition, as you said overriding the "fade in" value of the new entrance

Setting `globalCtx->transitionType = 0xFF` will reach `Fault_AddHungupAndCrash("../z_play.c", 2290);`,
so indeed it may be a bad idea to make 0xFF a `TRANS_TYPE_`

I would see some random name working though, like `TRANS_END_ENTRANCE_TYPE` as in "for the transition end use the entrance's type". I don't think there's really an elegant name, but it's only used in 4 spots so a comment actually explaining what's going on, and a name that can't be confused with something else, is enough I think

Reply:
with your explanation I understand what it means, but at first read `TRANS_END_USE_ENTRANCE_TYPE` was very hard to parse. You are right that its not used often, and not meant to be used by users of the system. But if we could have something a bit easier to read I think it would be nice

not that it affects the conversation much, but mm was given names for many save context fields, so the og name of this is `next_wipe`. So what we have currently was along the same lines as their intention. 
I feel like naming it after being an override still make things more clear, but Im not 100% set on that at the moment

But yea I do think naming it separate from `TRANS_TYPE_` is needed here cause it represents a different kind of number. Just need to decide on something to go with

Reply:
brainstorming here...

Maybe dropping the "END" makes it more legible `TRANS_USE_ENTRANCE_TYPE`

Or replace it with "NEXT"? `TRANS_NEXT_USE_ENTRANCE_TYPE`

`TRANS_NEXT_TYPE_USE_ENTRANCE` (also close to `nextTransitionType` 👀)

`TRANS_NEXT_TYPE_FROM_ENTRANCE`

Reply:
I feel like adding "override" in the name somehow would be sort of cramming it in there, and you would also have to negate it which usually doesn't look too good in names

I do like `TRANS_NEXT_TYPE_FROM_ENTRANCE` myself

I don't see the 0xFF value as "don't override" either but as the default behavior, since it's reset after being used if not 0xFF
https://github.com/fig02/oot/blob/01c573a3416aff4619b2446722aeb0cad7293792/src/code/z_play.c#L350-L357
```c
        if (gSaveContext.nextTransitionType == 0xFF) {
            // fade in
            globalCtx->transitionType =
                (gEntranceTable[((void)0, gSaveContext.entranceIndex) + tempSetupIndex].field >> 7) & 0x7F;
        } else {
            globalCtx->transitionType = gSaveContext.nextTransitionType;
            gSaveContext.nextTransitionType = 0xFF;
        }
```

Reply:
I also see as the default behavior, which is why I suggested `_NONE` originally. I would suggest `TRANS_NEXT_TYPE_DEFAULT` right now, but `TRANS_NEXT_TYPE_FROM_ENTRANCE` seems fine as well

Reply:
I think i like default much more. a comment can explain that default in this context means that it will take from the entrance table. but that info being in the define name itself seems like a lot to try and parse, imo

Reply:
+1 on `TRANS_NEXT_TYPE_DEFAULT`, the name indeed doesn't need to say where the default comes from